pinot-tools/src/main/java/org/apache/pinot/tools/filesystem/PinotFSBenchmarkDriver.java

```diff
@@ -0,0 +1,252 @@
+        File tmpFile = new File(tmpDirectory, relativePath);
+        tmpFile.createNewFile();
+        _pinotFS.copyFromLocalFile(tmpFile, combinePath(directoryUri, relativePath));
+      }
+      LOGGER.info("Took {} ms to create {} segments for directory_{}",
+          System.currentTimeMillis() - prepareTime, numSegments, i);
+      numSegments *= 10;
+    }
+
+    // reset numSegments
+    numSegments = 1;
+    for (int i = 0; i < numDirectories; i++) {
+      for (int j = 0; j < DEFAULT_RETRY; j++) {
+        URI directoryUri = combinePath(listTestUri, "directory_" + i);
+        long listFilesStart = System.currentTimeMillis();
+        String[] lists = _pinotFS.listFiles(directoryUri, true);
+        LOGGER.info("{}: took {} ms to listFiles. directory_{} ({} segments)", j,
+            System.currentTimeMillis() - listFilesStart, i, lists.length);
+        Preconditions.checkState(lists.length == numSegments);
+      }
+      numSegments *= 10;
+      LOGGER.info("");
+    }
+  }
+
+  private void testListFiles() throws Exception {
+    LOGGER.info("========= List Files ==========");
+    long prepareTime = System.currentTimeMillis();
+    URI listTestUri = combinePath(_baseDirectoryUri, "listTest");
+    _pinotFS.mkdir(listTestUri);
+    LOGGER.info("Created {} for list test...", listTestUri);
+
+    for (int i = 0; i < _numSegmentsForListFilesTest; i++) {
+      String relativePath = "segment_" + i;
+      File tmpFile = new File(_localTempDir.getPath(), relativePath);
+      tmpFile.createNewFile();
+      _pinotFS.copyFromLocalFile(tmpFile, combinePath(listTestUri, relativePath));
+    }
+    LOGGER.info("Took {} ms to create {} segments.",
+        System.currentTimeMillis() - prepareTime, _numSegmentsForListFilesTest);
+
+    for (int i = 0; i < DEFAULT_RETRY; i++) {
+      long listFilesStart = System.currentTimeMillis();
+      String[] lists = _pinotFS.listFiles(listTestUri, true);
+      LOGGER.info("{}: took {} ms to listFiles.", i, System.currentTimeMillis() - listFilesStart);
+      Preconditions.checkState(lists.length == _numSegmentsForListFilesTest);
+    }
+  }
+
+  private void testCopies() throws Exception {
+    LOGGER.info("\n========= Uploads and Downloads ==========");
+    URI copyTestUri = combinePath(_baseDirectoryUri, "copyFiles");
+    _pinotFS.mkdir(copyTestUri);
+    LOGGER.info("Created {} for copy test...", copyTestUri);
+
+    long fileSizeInBytes = _dataSizeInMBsForCopyTest * 1024 * 1024;
+    File largeTmpFile = createFileWithSize("largeFile", fileSizeInBytes);
+    for (int i = 0; i < DEFAULT_RETRY; i++) {
+      URI largeFileDstUri = combinePath(copyTestUri, largeTmpFile.getName() + "_" + i);
+      long copyStart = System.currentTimeMillis();
+      _pinotFS.copyFromLocalFile(largeTmpFile, largeFileDstUri);
+      LOGGER.info("{}: took {} ms to copyFromLocal, fileSize: {} MB.", i, System.currentTimeMillis() - copyStart,
+          _dataSizeInMBsForCopyTest);
+    }
+
+    LOGGER.info("");
```

Comment:
why do we have these empty log messages?

Reply:
removed